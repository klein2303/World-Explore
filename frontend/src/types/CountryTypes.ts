export type CountryType = {
    name: string;
    continent: string;
    capital: string;
    largest_city: string;
    currency: string;
    language: string;
    population: number;
    land_area: number;
    agriculture_area: string;
    forest_area: string;
    co2_emissions: number;
    image: string;
};
