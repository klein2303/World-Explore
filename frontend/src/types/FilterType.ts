export type FilterType = {
    continent: ContinentType;
    search: string;
    sort: string;
};

type ContinentType = {
    Africa: boolean;
    Asia: boolean;
    Europe: boolean;
    "North America": boolean;
    Oceania: boolean;
    "South America": boolean;
}