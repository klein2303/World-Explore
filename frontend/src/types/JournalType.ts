export type JournalTypeRead = {
    country: string;
    reviews: readReviewType[];
};

export type JournalTypeWrite = {
    country: string;
    reviews: writeReviewType[];
};

export type readReviewType = {
    title: string;
    date: string;
    rating: number;
    text: string;
    public: boolean;
    journal: {
        profile: {
            username: string;
        };
    };
};

export type writeReviewType = {
    title: string;
    date: string;
    rating: number;
    text: string;
    public: boolean;
    journal: {
        profile: {
            email: string;
        };
    };
}