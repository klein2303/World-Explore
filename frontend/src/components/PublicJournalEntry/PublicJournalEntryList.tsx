import { JournalTypeRead } from "../../types/JournalType";
import PublicJournalEntry from "./PublicJournalEntry";
import styles from "./PublicJournalEntryList.module.css";

type PublicJournalEntriesProps = {
    journal: JournalTypeRead;
};

const PublicJournalEntryList = ({ journal }: PublicJournalEntriesProps) => {
    const { country, reviews } = journal;

    return (
        <>
            {/* Grid for journal entries */}
            <article
                className={styles.entriesGrid}
                role="list"
                aria-description={`Public journal entries for ${country}`}>
                {reviews.map((review) => (
                    <div key={review.title} role="listitem">
                        <PublicJournalEntry review={review} />
                    </div>
                ))}
            </article>
        </>
    );
};

export default PublicJournalEntryList;
