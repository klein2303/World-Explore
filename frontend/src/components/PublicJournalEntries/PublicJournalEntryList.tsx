import {JournalType} from "../../types/JournalType"; // Import JournalType
import PublicJournalEntry from "./PublicJournalEntry"; // Import PublicJournalEntry component
import styles from "./PublicJournalEntryList.module.css"; // Import CSS module

type PublicJournalEntriesProps = {
    journal: JournalType; // Expecting a full JournalType as prop, containing country and reviews
};

const PublicJournalEntries: React.FC<PublicJournalEntriesProps> = ({journal}) => {
    const {country, reviews} = journal; // Destructure country and reviews from journal prop

    // Filter public reviews
    const publicReviews = reviews.filter((review) => review.public);

    if (publicReviews.length === 0) {
        return <p>No public journal entries available for {country}.</p>;
    }

    return (
        <section className={styles.journalContainer}>
            <h3 className={styles.journalTitle}>Public Journal Entries for {country}</h3>
            {/* Display average rating and stars here*/}

            {/* Horizontal line */}
            <div className={styles.horizontalLine}></div>

            <div className={styles.entriesGrid}>
                {publicReviews.map((review) => (
                    <PublicJournalEntry key={review.id} review={review} />
                ))}
            </div>
        </section>
    );
};

export default PublicJournalEntries;
