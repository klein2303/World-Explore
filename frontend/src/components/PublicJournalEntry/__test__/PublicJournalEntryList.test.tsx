import { render, screen } from "@testing-library/react";
import { describe, it, expect, vi } from "vitest";
import PublicJournalEntryList from "../PublicJournalEntryList";
import { JournalTypeRead } from "../../../types/JournalType";

// Mock the PublicJournalEntry component
vi.mock("../PublicJournalEntry", () => ({
    default: vi.fn(() => <div data-testid="mock-public-journal-entry">Mock Entry</div>),
}));

describe("PublicJournalEntryList Component", () => {
    const mockJournal: JournalTypeRead = {
        country: "Wonderland",
        reviews: [
            {
                title: "Amazing Experience",
                date: "2024-11-19",
                rating: 5,
                text: "This was an unforgettable adventure!",
                public: true,
                journal: {
                    profile: {
                        username: "user1",
                    },
                },
            },
            {
                title: "Not bad",
                date: "2024-11-18",
                rating: 3,
                text: "It was a decent experience overall.",
                public: false,
                journal: {
                    profile: {
                        username: "user2",
                    },
                },
            },
        ],
    };

    it("renders correctly and matches snapshot", () => {
        const { asFragment } = render(<PublicJournalEntryList journal={mockJournal} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("displays the correct number of PublicJournalEntry components", () => {
        render(<PublicJournalEntryList journal={mockJournal} />);
        const entries = screen.getAllByTestId("mock-public-journal-entry");
        expect(entries).toHaveLength(mockJournal.reviews.length);
    });

    it("renders the grid with the role 'list'", () => {
        render(<PublicJournalEntryList journal={mockJournal} />);
        const articleElement = screen.getByRole("list");
        expect(articleElement).toBeInTheDocument();
    });
});
