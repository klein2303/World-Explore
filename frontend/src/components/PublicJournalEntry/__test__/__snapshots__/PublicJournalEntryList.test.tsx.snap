// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PublicJournalEntryList Component > renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <article
    aria-description="Public journal entries for Wonderland"
    class="_entriesGrid_bf4d47"
    role="list"
  >
    <div
      role="listitem"
    >
      <div
        data-testid="mock-public-journal-entry"
      >
        Mock Entry
      </div>
    </div>
    <div
      role="listitem"
    >
      <div
        data-testid="mock-public-journal-entry"
      >
        Mock Entry
      </div>
    </div>
  </article>
</DocumentFragment>
`;
